rc/extra/editorconfig.kak

```diff
@@ -1,35 +1,61 @@
-def editorconfig-load -docstring "Set indentation options according to editorconfig file" %{
+# http://editorconfig.org/#file-format-details
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+decl -hidden bool editorconfig_trim_trailing_whitespace false
+decl -hidden int  editorconfig_insert_final_newline     0
+
+def -hidden _editorconfig-bufwritepre-hook %{
+    hook buffer BufWritePre %val{buffile} -group editorconfig-hooks %{ %sh{
+        if [ ${kak_opt_editorconfig_trim_trailing_whitespace} = "true" ]; then
+            printf %s\\n "try %{ exec -draft %{ %s\h+$<ret>d } }"
+        fi
+        if [ ${kak_opt_editorconfig_insert_final_newline} -gt 0 ]; then
+            printf %s\\n "try %{ exec -draft %{ %s\v+\Z<ret>d } }"
+        fi
+        if [ ${kak_opt_editorconfig_insert_final_newline} -gt 1 ]; then
+            printf %s\\n "try %{ exec -draft %{ geo<esc> } }"
+        fi
+    } }
+}
+
+def editorconfig-load -docstring "Set indentation options and whitespace behavior according to editorconfig" %{
     %sh{
         command -v editorconfig >/dev/null 2>&1 || { echo 'echo -color Error The editorconfig tool could not be found'; exit 1; }
-        editorconfig $kak_buffile | awk -F= -- '
-            /indent_style=/ { indent_style = $2 }
-            /indent_size=/  { indent_size = $2 == "tab" ? 4 : $2  }
-            /tab_width=/    { tab_width = $2 }
-            /end_of_line=/  { end_of_line = $2 }
-            /charset=/      { charset = $2 }
+        editorconfig "${kak_buffile}" | awk -F= -- '
```

Comment:
The reason for a global `.editorconfig` instead of putting it in `kakrc` is because `.editorconfig` will work for a variety of editors, not just Kakoune, and so any editor I use will pick up the same configuration.

When you say this script implements the same behavior as the official binary, that doesn't actually make any sense. We're talking here about scratch buffers, which do not have a corresponding file on disk. The official binary is only capable of operating given a file path on disk. Therefore, the official binary has no idea about the concept of scratch buffers, and therefore does not provide any opinion either way as to how scratch buffers should behave.